### PR TITLE
refactor telemetry for different host names.

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
update mapbox telemetry lib version.

### CODE REVIEW

**Comment by vladzima** [comment]
Linked issue?

**Comment by alexander-kulikovskii** [comment]
Closes [1412](https://app.zenhub.com/workspaces/vision-sdk-5ca52e6c027e3e7234ac7512/issues/mapbox/mapbox-vision/1412)

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+            }
+
+            currentCountry = country
+
+            if (isRecording) {
+                return
+            }
+
+            when {
+                currentCountry == Country.Unknown && country != Country.Unknown -> {
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: currentCountry == country at that point

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+        companion object {
+            private const val CACHE_DIR = "Cache"
+            private const val VIDEO_BUFFERS_DIR = "Buffers"
+            private const val DIR_TELEMETRY = "Telemetry"
+        }
+
+        private lateinit var sessionWriter: SessionWriter
+        private var currentCountry = Country.Unknown
+
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: remove extra lines

File: MapboxVision/src/main/java/com/mapbox/vision/sync/metagenerator/TelemetryMetaGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        saveDirPath: String,
+        startRecordCoreMillis: Long
+    ) {
+        val arr = JSONArray()
+        for (videoPart in videoClipMap) {
+            val jsonPath = JSONObject()
+            val paths = videoPart.key.split("/")
+            val name = paths[paths.size - 1]
+
+            val startSeconds = startRecordCoreMillis / 1000f + videoPart.value.startSeconds
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/metagenerator/TelemetryMetaGenerator.kt]:
```suggestion
            val startSeconds = TimeUnit.MILLS.toSecond(startRecordCoreMillis) + videoPart.value.startSeconds
```

and next line

File: MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    private val videoRecorder: VideoRecorder,
+    private val telemetryImageSaverImpl: TelemetryImageSaverImpl,
+    private val sessionWriterListener: SessionWriterListener
+) : SessionWriter {
+
+    companion object {
+        private val SESSION_LENGTH_MILLIS = TimeUnit.MINUTES.toMillis(5)
+    }
+
+    private val handler = WorkThreadHandler("Session")
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt]:
```suggestion
    private val workingHandler = WorkThreadHandler("Session")
```
it's not a handler

File: MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            stopSession()
+        }
+    }
+
+    private fun startSession() {
+        videoRecorder.startRecording(buffers.getBuffer())
+        nativeVisionManager.startTelemetrySavingSession(sessionCacheDir)
+        telemetryImageSaverImpl.start(sessionCacheDir)
+
+        startRecordCoreMillis = (nativeVisionManager.getCoreTimeSeconds() * 1000).toLong()
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt]:
```suggestion
        startRecordCoreMillis = TimeUnit.SECONDS.toMills(nativeVisionManager.getCoreTimeSeconds().toLong)
```

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+        private val nativeVisionManager: NativeVisionManager,
+        private val videoRecorder: VideoRecorder,
+        private val mapboxTelemetry: MapboxTelemetry,
+        private val telemetryImageSaver: TelemetryImageSaver
+    ) : SessionManager {
+
+        companion object {
+            private const val CACHE_DIR = "Cache"
+            private const val VIDEO_BUFFERS_DIR = "Buffers"
+            private const val DIR_TELEMETRY = "Telemetry"
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: rename to TELEMETRY_DIR to be consistent

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 7]
```diff
@@ -76,12 +77,11 @@ object VisionManager : BaseVisionManager {
...
     private lateinit var videoRecorder: VideoRecorder
 
-    private var isRecording = false
+    private lateinit var mapboxTelemetry: MapboxTelemetry
 
     private val handlerMain = Handler(Looper.getMainLooper())
 
-    // TODO temporary fix, this should be moved to core
-    private var currentCountry: Country = Country.Unknown
+    private var isTurnstileEventSent = false
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: @RingerJK could you clarify we still need to send turnstile events?

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 8]
```diff
@@ -201,6 +186,33 @@ object VisionManager : BaseVisionManager {
 
         this.videoSource = videoSource
         this.videoRecorder = videoRecorder
+
+        mapboxTelemetry = MapboxTelemetry(
+            VisionManager.application,
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: just `application` here and below

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 9]
```diff
@@ -218,32 +230,11 @@ object VisionManager : BaseVisionManager {
...
-                    when (country) {
-                        Country.China -> {
-                            sessionManager.stop()
-                        }
-                        Country.Unknown, Country.UK, Country.USA, Country.Other -> {
-                            sessionManager.start()
-                        }
-                    }
-                }
+                sessionManager.setCountry(country)
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: Is `handlerMain.post(sessionManager::setCountry)` possible here?

File: MapboxVision/src/main/java/com/mapbox/vision/sync/syncmanager/TelemetrySyncManager.kt [hunk 10]
```diff
@@ -0,0 +1,279 @@
+package com.mapbox.vision.sync.syncmanager
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/syncmanager/TelemetrySyncManager.kt]: Was it changed from `Impl` or just factored out here?

File: MapboxVision/src/main/java/com/mapbox/vision/utils/FileUtils.kt [hunk 11]
```diff
@@ -34,4 +34,10 @@ internal object FileUtils {
 
         return "${getAbsoluteDir(dir.absolutePath)}/"
     }
+
+    fun moveFilesFromOneDirToAnother(fromDirPath: String, toDirPath: String) {
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/utils/FileUtils.kt]: Use `renameTo` for consistency with `File` api?

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt (hunk 7 above)
Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: yep, analytic-team use it for audit and some other issues

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt (hunk 1 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: thx

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt (hunk 2 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: done

File: MapboxVision/src/main/java/com/mapbox/vision/sync/metagenerator/TelemetryMetaGenerator.kt (hunk 3 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/metagenerator/TelemetryMetaGenerator.kt]: refactor

File: MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt (hunk 4 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt]: refactor

File: MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt (hunk 5 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/session/RotatedBuffersSessionWriter.kt]: refactor

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt (hunk 6 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: refactor

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt (hunk 8 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: refactor

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt (hunk 9 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: not possible. leave current version

File: MapboxVision/src/main/java/com/mapbox/vision/utils/FileUtils.kt (hunk 11 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/FileUtils.kt]: refactor

File: MapboxVision/src/main/java/com/mapbox/vision/sync/syncmanager/TelemetrySyncManager.kt (hunk 10 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/syncmanager/TelemetrySyncManager.kt]: try to leave as is.
please check

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 12]
```diff
@@ -201,6 +186,34 @@ object VisionManager : BaseVisionManager {
...
+            isTurnstileEventSent = true
+        }
+
+        nativeVisionManager.create(
+            telemetryEventManager = MapboxTelemetryEventManager(mapboxTelemetry),
+            telemetryImageSaver = telemetryImageSaver
+        )
+
+        sessionManager = SessionManager.Impl(
+            application, nativeVisionManager,
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]:
```suggestion
            application, 
            nativeVisionManager,
```

`nativeVisionManager` to next line

File: MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+        private val sessionWriterListener = object : SessionWriterListener {
+            override fun onSessionStop(
+                clips: Array<VideoClip>,
+                videoPath: String,
+                outputPath: String,
+                sessionStartMillis: Long
+            ) {
+                if (currentCountry != Country.Unknown) {
+
+                    val sessionCountryDir = getCountryTelemetryDir()
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]:
```suggestion
                    val sessionCountryDir = getCountryTelemetryDir() ?: return
```
and remove `.let{ }` block

Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt]: done

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt (hunk 12 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: done

**Pull request merged**